src/core/src/bound_evaluate.cpp

```diff
@@ -80,8 +80,17 @@ ov::Tensor evaluate_bound(const Output<Node>& output, bool is_upper, bool invali
             ov::TensorVector outputs;
             for (const auto& out : node->outputs()) {
                 const auto& out_shape = out.get_partial_shape();
-                auto shape = out_shape.is_static() ? out_shape.to_shape() : Shape(out_shape.rank().get_length());
-                outputs.emplace_back(out.get_element_type(), shape);
+                const auto& out_et = out.get_element_type();
+
+                if (out_et.is_dynamic()) {
+                    outputs.emplace_back();
+                } else if (out_shape.is_static()) {
+                    outputs.emplace_back(out_et, out_shape.to_shape());
+                } else if (out_shape.rank().is_static()) {
+                    outputs.emplace_back(out_et, Shape(out_shape.rank().get_length()));
+                } else {
+                    outputs.emplace_back(out_et, Shape{0, std::numeric_limits<size_t>::max()});
```

Comment:
not clear why you create Shape like for 1D tensor if the rank is dynamic? Please clarify

Reply:
Or how we (the logic below) will distinguish it from a case of 2d tensor of shape `0, std::numeric_limits<size_t>::max()`

Reply:
Yes true, changed to Shape{0} as is used as special case to convert to dynamic tensor.

Reply:
@jane-intel, what do you think how we should handle this case?

Reply:
I think that if output shape of the operation is yet to be known -- we may let it be scalar and perform proper memory allocation later -- during evaluate method we would collect exact number of elements in the output and will set output shape. So, I agree with the solution.